e2e/nomostest/registry_server.go

```diff
@@ -114,22 +114,28 @@ func setupRegistryClient(nt *NT, opts *ntopts.New) error {
 	if opts.RequireOCIProvider && *e2e.OCIProvider == e2e.Local || opts.RequireHelmProvider && *e2e.HelmProvider == e2e.Local {
 		nt.portForwardRegistryServer(opts.RequireHelmProvider, opts.RequireOCIProvider)
 	}
-	// For non-local registries, just login before each test and logout after.
-	if opts.RequireOCIProvider && *e2e.OCIProvider != e2e.Local {
+	// Login before each test, then reset the registry and logout after the test.
```

Comment:
Given that the login/logout is idempotent, I wrote this to just allow redundant login for local registries as it keeps the control flow simpler